### PR TITLE
Upgrade to v2.5.0

### PR DESCRIPTION
**Pull request by metinavc1**
SCLI-62 Internal Printing Enhancements
SCLI-63 Locking Mechanism for Pay at Table
SCLI-64 Suppress Merchant Password for MOTO
SCLI-65 Get Open Tables API
SCLI-67 Issue with receipt header on VAA 2.5+

### CODE REVIEW

File: Library/Model/SPIPreAuth.m [hunk 1]
```diff
@@ -147,9 +180,15 @@ - (void)initiatePartialCancellationTx:(NSString *)posRefId
 - (void)initiateExtendTx:(NSString *)posRefId
                preauthId:(NSString *)preauthId
               completion:(SPICompletionTxResult)completion {
+}
```
Comment (chrishulbert) [inline Library/Model/SPIPreAuth.m:187]: Should this method be empty?

File: Library/SPIClient.h [hunk 2]
```diff
@@ -403,6 +480,17 @@ isSuppressMerchantPassword:(BOOL)isSuppressMerchantPassword
 - (void)initiateSettlementEnquiry:(NSString *)posRefId
                        completion:(SPICompletionTxResult)completion;
 
+/**
+ Initiates a Mail Order / Telephone Order purchase transaction.
```
Comment (chrishulbert) [inline Library/SPIClient.h]: Was this comment copied from a MOTO function and forgot to get changed?

File: Library/SPIClient.m [hunk 3]
```diff
@@ -978,14 +1054,14 @@ - (void)setTestMode:(BOOL)testMode {
 
 - (BOOL)hasSerialNumberChanged:(NSString *)updatedSerialNumber {
     if ([_serialNumber isEqualToString:updatedSerialNumber])
-        return false;
+    return false;
```
Comment (chrishulbert) [inline Library/SPIClient.m]: This would be clearer if the body of the 'if' statement had braces {}. Eg this is how heartbleed happened :)

**Review by chrishulbert — CHANGES_REQUESTED** [review]
Cheers 🍻

File: Library/SPIClient.m (hunk 3 above)
Comment (metinavc1) [reply Library/SPIClient.m]: it is fixed

File: Library/Model/SPIPreAuth.m (hunk 1 above)
Comment (metinavc1) [reply Library/Model/SPIPreAuth.m:187]: it shouldn't be empty. Thanks.

File: Library/SPIClient.h (hunk 2 above)
Comment (metinavc1) [reply Library/SPIClient.h]: it's fixed

**Review by chrishulbert — APPROVED** [review]

**Pull request merged**